### ISSUE TITLE
Personal information leak

### ISSUE THREAD

**Issue by xWiiLLz** [foundjem/log3000-TP4#1]
Hi @foundjem,

Pretty sure this was not intentional, but [a commit in this repository](https://github.com/foundjem/log3000-TP4/commit/0b3288ef1f602ca5b3fee716551ff8c52ca55054#diff-e5564bfed4ce70cce598b77dd61e9fe0R107) introduced your personal file history in the sublime text workspace.

Perhaps you should remove it?

Hope this helps

**Comment by foundjem** [comment]
Hello Williams,
you are right. I removed it :)

Thank you!
Armstrong

**Comment by xWiiLLz** [comment]
My pleasure!

Also, since you only commited again to remove it, it's still present in the git history (anyone can have acces to it if they checkout commit #0b3288ef1f602ca5b3fee716551ff8c52ca55054). I would suggest using a git rebase, or checking out before and making a force push.

Regarding the TP4, what is expected for us to do? I see a lot of Dockerfiles that are useless. Should we go ahead and delete those?

Also, the dotnet projects seem to reference a MySQL database, but the docker-compose at the root level includes a postgres DB. Are we expected to edit the C# code to match the docker-compose file and use a PostgreSQL database instead, or should we rather  edit the docker-compose file appropriately to fill out what's missing?

Thanks a lot,
William

**Comment by foundjem** [comment]
Hello Williams,
Sounds good!

> On Apr 2, 2020, at 23:02, William Bergeron-Drouin <notifications@github.com> wrote:
> 
> ﻿
> My pleasure!
> 
> Also, since you only commited again to remove it, it's still present in the git history (anyone can have acces to it if they checkout commit #0b3288ef1f602ca5b3fee716551ff8c52ca55054). I would suggest using a git rebase, or checking out before and making a force push.
> 
> Regarding the TP4, what is expected for us to do? I see a lot of Dockerfiles that are useless. Should we go ahead and delete those?
> 
The purpose of this TP is to make sure students can identify how a complex IaC system works. In real life applications, you will see things like this, and a lot messy. Now, as Engineers you are expected to demonstrate best practices. But, make sure they system work first before  you attempt to delete anything.

> Also, the dotnet projects seem to reference a MySQL database, but the docker-compose at the root level includes a postgres DB. Are we expected to edit the C# code to match the docker-compose file and use a PostgreSQL database instead, or should we rather edit the docker-compose file appropriately to fill out what's missing?
> 
Yes, i expect you to edit things and make strong suggestions that works. Always justify your rationale. If you can implement something new, or  criticize the entire applications, you are more than welcome to do so. However, your justifications should be strong enough to convince me.

Thanks,
Armstrong

**Comment by xWiiLLz** [comment]
> Yes, i expect you to edit things and make strong suggestions that works. Always justify your rationale. If you can implement something new, or  criticize the entire applications, you are more than welcome to do so. However, your justifications should be strong enough to convince me.

There are multiple versions of both the **Worker** app (java/dotnet) and the **Result** app (Node/Dotnet). While I'm able to fix them and get them all to work, what is expected of us? Do we get to decide which app we want to maintain and adapt for best practices?

For example, could we stick to the Java Worker and ditch the Dotnet worker? I can get the Java worker to work pretty easily since it's using both Postgres and Redis, but for the dotnet one, it's a lot of work:

1. Changing EF Core to use Postgres instead of MySQL (which most likely will require an update to dotnet 3.1)
2. Changing NATS to use Redis, since all of our docker-compose reffer to Redis

Per my understanding, the goal of this TP is to make sense out of this stack and have a working production-ready deployment using the provided services. I would very much like to be able to choose whichever I find best for scaling and maintaining, then removing all the others that are just plain noise at this point.


Let me know if my understanding is correct,

Thanks,
William

**Comment by foundjem** [comment]
> On Apr 3, 2020, at 12:31 PM, William Bergeron-Drouin <notifications@github.com> wrote:
> 
> 
> Yes, i expect you to edit things and make strong suggestions that works. Always justify your rationale. If you can implement something new, or criticize the entire applications, you are more than welcome to do so. However, your justifications should be strong enough to convince me.
> 
> There are multiple versions of both the Worker app (java/dotnet) and the Result app (Node/Dotnet). While I'm able to fix them and get them all to work, what is expected of us? Do we get to decide which app we want to maintain and adapt for best practices?
> 
> 
Yes, I mean, I am tolerant and will consider those who make great efforts, PLEASE! Document and explain you work.
> For example, could we stick to the Java Worker and ditch the Dotnet worker? I can get the Java worker to work pretty easily since it's using both Postgres and Redis, but for the dotnet one, it's a lot of work:
> 
> 
Be careful not to say this “It’s a lot of work” in your report, it might hurts your marks. 
> Changing EF Core to use Postgres instead of MySQL (which most likely will require an update to dotnet 3.1)
> Changing NATS to use Redis, since all of our docker-compose reffer to Redis

Good suggestions just have to give more details in your report.
> Per my understanding, the goal of this TP is to make sense out of this stack and have a working production-ready deployment using the provided services. I would very much like to be able to choose whichever I find best for scaling and maintaining, then removing all the others that are just plain noise at this point.
> 
> 
Sure! But be careful that the end product must satisfy a basic and standard use case of a voting app. Don’t over complicate things.
> Let me know if my understanding is correct,
> 
> 
You are on the right track, keep riding on.

Regards,
Armstrong
